cmake/Modules/FindMxnet.cmake

```diff
@@ -28,6 +28,10 @@ if (LEN EQUAL "5")
     string(TOUPPER ${Mxnet_USE_MKLDNN} Mxnet_USE_MKLDNN)
     string(TOUPPER ${Mxnet_USE_CUDA} Mxnet_USE_CUDA)
     if (Mxnet_USE_MKLDNN)
+        if (NOT EXISTS ${Mxnet_INCLUDE_DIRS}/mkldnn)
+            set(MXNET_FOUND FALSE)
```

Comment:
You want something like that:
```
set(MXNET_FOUND FALSE)
if (MXNET_FIND_REQUIRED)
    set(MSG_LEVEL "FATAL_ERROR")
else()
    set(MSG_LEVEL "WARNING")
endif()
message(${MSG_LEVEL} "MXNet was found with mkl-dnn support but mkldnn header files are missing. Please, install MXNet with mkldnn header files.")
return()
```

Reply:
Why is that not handled by find_package? According to doc: 

> The package configuration file may set <PackageName>_FOUND to false to tell find_package that component requirements are not satisfied.

https://cmake.org/cmake/help/latest/command/find_package.html